lib/ecto/changeset.ex

```diff
@@ -604,6 +610,15 @@ defmodule Ecto.Changeset do
       iex> changeset.params
       %{}
 
+  You can define a custom error message function.
+
+      iex> params = %{title: 1, body: ""}
+      iex> custom_errors = [title: "must be a string"]
+      iex> error_msg_func = fn field, meta -> custom_errors[field] || meta[:message]
```

Comment:
```suggestion
      iex> error_msg_func = fn field, meta -> custom_errors[field] || meta[:message] end
```

Reply:
Although instead of passing the message, I would suggest to allow the function to return nil and, if nil is returned, it uses the default one?

Reply:
thanks that's much nicer. updated.